Added callback for viewport modification - allowing tests for respons…

maqqju opened the pull request:
Added callback for viewport modification - allowing tests for responsiveness.

maqqju committed:
Added callback for viewport modification - allowing tests for responsiveness.

maqqju committed:
Added documentation to change supporting viewportSize configuration.

maqqju commented:
Build was already broken beforehand.

nathanboktae commented on mocha-phantomjs-core.js:
```diff
@@ -114,6 +114,15 @@ page.onCallback = function(data) {
     } else if (typeof data.screenshot === 'string') {
       page.render(data.screenshot + '.png')
+    } else if (data.viewportSize && (data.viewportSize.x || data.viewportSize.y)) {
+      setTimeout(function() {
```
Why is this done asynchronously?

nathanboktae commented on mocha-phantomjs-core.js:
```diff
@@ -114,6 +114,15 @@ page.onCallback = function(data) {
+            width : _x || _current.width,
+            height : _y || _current.height
+          };
+        })(page, page.viewportSize, data.viewportSize.x, data.viewportSize.y);
```
Just simply do

``` javascript
page.viewportSize = {
  width: data.viewportSize.width || page.viewportSize.width,
  height: data.viewportSize.width || page.viewportSize.height,
}
```

maqqju replied:
Strange that first time I had tried it this way, it was not working - but then had fixed it with `setTimeout`.

Tried it again - and it's fine now.

nathanboktae commented on README.md:
```diff
@@ -116,6 +116,23 @@ if (window.callPhantom) {
+  window.callPhantom({
+    viewportSize : {
+      x : 100,
+      y : 100
```
Using `x` and `y` here when `width` and `height` is used on the API level is confusing. Change it to `width` and `height` and mention they are optional.

nathanboktae commented:
Add a test too please. It's probably not supported on slimerjs, but if that's the case, it's ok.

maqqju committed:
Pull request feedback :
1) Changed x, y to width, height
2) Removed setTimeout, and simplified viewport changing
3) Wrote tests - tested on both phantomjs and slimerjs

maqqju commented:
Changed according to feedback, added tests.

nathanboktae commented on test/lib/viewport-size.js:
```diff
@@ -0,0 +1,23 @@
+  it('changes viewport size', function() {
+    if (window.callPhantom) {
+    	console.log("Changing viewport size to : width 500, height 500")
+    	callPhantom({'viewportSize': { width : 500, height : 500}})
```
You need to assert the changes in window size.

nathanboktae commented:
Thanks @maqqju , add those assertions to the tests and I'll take the PR. I need to fix slimerjs, and it's ok if this doesn't work there (there is actually a window with slimerjs as it is not truly headless).

maqqju committed:
Added assertions. Test works on both SlimerJS and PhantomJS

maqqju commented:
@nathanboktae - assertions added, and tested on both SlimerJS and PhantomJS. Thanks a lot for your feedback.

nathanboktae commented:
Thanks, one more thing - you need to wire up your tests in `core.tests.coffee` like this: https://github.com/nathanboktae/mocha-phantomjs-core/blob/master/test/core.tests.coffee#L269-L275

maqqju commented:
Thanks @nathanboktae 

I am having problems to run `npm test`. I read #53 and pointed the `core.coffee.tests` to my `phantomjs` and `slimerjs` instances - only to have more different errors.

nathanboktae commented:
the tests expect `phantomjs` and `slimerjs` to be at the root of the repo. so just copy your phantomjs binary there like CI does.

maqqju committed:
Wired viewport-size tests in core.tests.coffee

maqqju commented:
Managed to make tests work - 2 problem were found:
- for some reason the `file:///` protocol was not being appended to the first part of the `fileURL`
- `fileURL` was not returning the path

Did not commit my fixes as I am not sure whether this is a misconfiguration on my end. If need be I can insert them again.

nathanboktae commented:
> Did not commit my fixes as I am not sure whether this is a misconfiguration on my end. 

No [the test worked](https://travis-ci.org/nathanboktae/mocha-phantomjs-core/jobs/127781359#L263-L264). Thanks for the PR.!
